trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmAccessionImporter.php

```diff
@@ -0,0 +1,885 @@
+    // where the file is located on the server.
+    $file_path = $arguments['files'][0]['file_path'];
+    if (!file_exists($file_path)) {
+      throw new \Exception(
+        t("File does not exist: @file", ['@file' => $file_path])
+      );
+    }
+
+    // Grab the genus name
+    $genus_name = $arguments['run_args']['genus_name'];
+
+    // Make sure our CVterms are all set
+    $this->setUpCVterms();
+
+    // Check if the stock_synonym table exists before moving forward
+    if (!$this->connection->schema()->tableExists('stock_synonym')) {
+      throw new \Exception(
+        t("Could not find stock_synonym table in the current database schema.")
+      );
+    }
+
+    // Set up the ability to track progress so we can report it to the user
+    $filesize = filesize($file_path);
+    $this->setTotalItems($filesize);
+    $this->setItemsHandled(0);
+    $bytes_read = 0;
+    $line_count = 0;
+
+    // Open the file and start iterating through each line
+    $GERMPLASM_FILE = fopen($file_path, 'r');
+    if(!$GERMPLASM_FILE) {
+      throw new \Exception(
+        t("Could not open file: @file", ['@file' => $file_path])
+      );
+    }
+
+    while (!feof($GERMPLASM_FILE)){
+      $current_line = fgets($GERMPLASM_FILE);
+      $line_count++;
+
+      // Calculate how many bytes we have read from the file and let the
+      // importer know how many have been processed so it can provide a
+      // progress indicator.
+      $bytes_read += mb_strlen($current_line);
+      $this->setItemsHandled($bytes_read);
+
+      // Check for empty lines, comment lines and a header line
+      $current_line = trim($current_line);
+      if ($current_line == '') continue;
+      if (preg_match('/^#/', $current_line)) continue;
+      if (preg_match('/^Germplasm/i', $current_line)) continue;
+
+      // Split our columns into an array for easier processing
+      $germplasm_columns = explode("\t", $current_line);
+      $num_columns = count($germplasm_columns);
+      if (count($germplasm_columns) < 4) {
+        $this->logger->error("Insufficient number of columns detected (<4) for line # @line", ['@line' => $line_count]);
+        $this->error_tracker = TRUE;
+				// Continue to next line since we already know this will cascade into
+				// further errors
+				continue;
+      }
+
+      // Collect our values from our current line into variables
+      // Since the 1st 4 columns are required, make sure there are values there
+      for($i=0; $i<4; $i++) {
+        $column = $i+1;
+        if ($germplasm_columns[$i] == '') {
+          $this->logger->error("Column @column is required and cannot be empty for line # @line", ['@column' => $column, '@line' => $line_count]);
+          $this->error_tracker = TRUE;
+					// Continue to next line since we already know this will cascade into
+					// further errors
+					continue 2;
+        }
+      }
+      $germplasm_name = $germplasm_columns[0];
```

Comment:
You'll want to add trim() around all of these ;-p 
```suggestion
      $germplasm_name = trim($germplasm_columns[0]);
```